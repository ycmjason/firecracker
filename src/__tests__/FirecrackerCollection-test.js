jest.mock('../FirecrackerDocument');
import { firestore } from './helpers/firebase';
import { FirecrackerCollection, FirecrackerDocument } from '..';

describe('FirecrackerCollection', () => {
  const $collection = firestore.collection('test');

  beforeAll(async () => {
    await $collection.doc('a').set({ value: 30 });
  });

  describe('new FirecrackerCollection($collection)', () => {
    it('should point to this.$collection the correct collection', () => {
      const collection = new FirecrackerCollection($collection);
      expect(collection.$collection).toBe($collection);
    });
  });

  describe('firecrackerCollection.findById(id)', () => {
    it('should call FirecrackerDocument.from($docRef)', async () => {
      const collection = new FirecrackerCollection($collection);
      FirecrackerDocument.from.mockResolvedValue('yoyo');

      expect(await collection.findById('a'))
        .toBe('yoyo');
      expect(FirecrackerDocument.from)
        .toHaveBeenCalledWith($collection.doc('a'));
    });
  });
});
