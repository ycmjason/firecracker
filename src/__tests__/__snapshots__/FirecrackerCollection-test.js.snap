// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FirecrackerCollection firecrackerCollection.createWithId(doc) should throw if id already exists 1`] = `[Error: col.the id already exists]`;
